subprojects/docs/src/docs/userguide/jvm/toolchains.adoc

```diff
@@ -399,17 +404,36 @@ So Gradle picks the first match in this order: Microsoft JDK 17.0.1.
-By injecting the `JavaToolchainService` in the plugin or task, it is also possible to wire a convention in those properties by obtaining the `JavaToolchainSpec` from the `java` extension on the project.
+When creating a plugin or a task that uses toolchains it is important to provide sensible defaults, as well as to allow users to override them.
+
+For JVM projects it is usually safe to assume that `java` plugin has been applied to the project.
+In such a case, it is appropriate to use the toolchain defined via the `java` extension as a default value for the tool property.
+This way the users will need to configure the toolchain only once on the project level.
+
 The example below showcases how to use the default toolchain as convention while allowing users to individually configure the toolchain per task.
 
 ====
-include::sample[dir="snippets/java/toolchain-task/kotlin/",files="build.gradle.kts"]
-include::sample[dir="snippets/java/toolchain-task/groovy/",files="build.gradle"]
+include::sample[dir="snippets/java/toolchain-task/kotlin/",files="build.gradle.kts[tags=custom-toolchain-task-with-java]"]
+include::sample[dir="snippets/java/toolchain-task/groovy/",files="build.gradle[tags=custom-toolchain-task-with-java]"]
 ====
+<1> We declare a `JavaLauncher` property on the task.
+It is important that the property is marked as a <<more_about_tasks.adoc#sec:task_input_nested_inputs,`@Nested` input>> to make sure the task is responsive to toolchain changes.
+<2> We obtain the toolchain spec from the `java` extension to use it as a default.
+<3> Using the `JavaToolchainService` we get a provider of the `JavaLauncher` that matches the toolchain.
+<4> Finally, we wire the launcher provider as a convention for our property.
+
+In a project where `java` plugin was applied we can use the task as follows:
 
-[NOTE]
 ====
-With the property correctly configured as `@Nested`, it will automatically track the Java major version, the vendor (if specified) and implementation (if specified) as an input.
+include::sample[dir="snippets/java/toolchain-task/kotlin/",files="build.gradle.kts[tags=custom-toolchain-task-with-java-usage]"]
+include::sample[dir="snippets/java/toolchain-task/groovy/",files="build.gradle[tags=custom-toolchain-task-with-java-usage]"]
 ====
+<1> The toolchain defined on the `java` extension is used by default to resolve the launcher.
+<2> The custom task without additional configuration will use the default Java 8 toolchain.
+<3> The other task overrides the value of the launcher by selecting a different toolchain using `javaToolchains` service.
+
+In a case when a task needs access to toolchains without `java` plugin being applied the toolchain service can be used directly.
+If an <<#sec:configuring_toolchain_specifications, unconfigured>> toolchain spec is provided to the service it will always return a tool provider for the toolchain that is running Gradle.
```

Comment:
```suggestion
If an <<#sec:configuring_toolchain_specifications, unconfigured>> toolchain spec is provided to the service, it will always return a tool provider for the toolchain that is running Gradle.
```